### ISSUE TITLE
DELETE statement from executed in MySQL doesn't executed in starrocks.

### ISSUE THREAD

**Issue by adhasahar97** [StarRocks/starrocks-connector-for-apache-flink#155]
When I delete rows in MySQL, the data still present in StarRocks.

Flink version: 1.15
MySQL CDC version: 2.2.1
StarRocks connector version: 1.2.4

**Comment by adhasahar97** [comment]
I can see the transaction is captured by StarRocks connector but the DELETE statement itself is not executed:
`Stream load completed, label : 21743297-4164-44f1-914e-bc3d608a2a5c, database : staging, table : crm_activity, body : {
    "TxnId": 14643,
    "Label": "21743297-4164-44f1-914e-bc3d608a2a5c",
    "Status": "Success",
    "Message": "OK",
    "NumberTotalRows": 1,
    "NumberLoadedRows": 1,
    "NumberFilteredRows": 0,
    "NumberUnselectedRows": 0,
    "LoadBytes": 453,
    "LoadTimeMs": 713,
    "BeginTxnTimeMs": 0,
    "StreamLoadPlanTimeMs": 0,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 692,
    "CommitAndPublishTimeMs": 20
}`

**Comment by banmoy** [comment]
@adhasahar97 Can you post the connector configuration?

**Comment by adhasahar97** [comment]
@banmoy here
`CREATE TABLE IF NOT EXISTS `default_catalog`.`staging`.`accounts_src` (
  `id` BIGINT NOT NULL,
  `email` STRING NULL,
  `phone` STRING NOT NULL,
  `displayname` STRING NOT NULL,
  `password` STRING NULL
  PRIMARY KEY(`id`)
 NOT ENFORCED
) with (
  'server-id' = '5501',
  'connector' = 'mysql-cdc',
  'hostname' = 'salesconnection-staging.cccdmxcdsrgz.ap-southeast-1.rds.amazonaws.com',
  'port' = '3306',
  'username' = 'propconadmin',
  'password' = 'kf7vm1od64ka9',
  'database-name' = 'staging',
  'table-name' = 'accounts'
);

CREATE TABLE IF NOT EXISTS `default_catalog`.`staging`.`accounts_sink` (
  `id` BIGINT NOT NULL,
  `email` STRING NULL,
  `phone` STRING NOT NULL,
  `displayname` STRING NOT NULL,
  `password` STRING NULL
  PRIMARY KEY(`id`)
 NOT ENFORCED
) with (
  'sink.properties.strip_outer_array' = 'true',
  'password' = '',
  'load-url' = 'starrockscluster-sample2-fe-service.default:8030',
  'username' = 'root',
  'sink.properties.format' = 'json',
  'sink.buffer-flush.interval-ms' = '15000',
  'jdbc-url' = 'jdbc:mysql://starrockscluster-sample2-fe-service.default:9030',
  'connector' = 'starrocks',
  'database-name' = 'staging',
  'table-name' = 'accounts',
  'sink.max-retries' = '10'
);

INSERT INTO `default_catalog`.`staging`.`accounts_sink` SELECT * FROM `default_catalog`.`staging`.`accounts_src`;`

**Comment by adhasahar97** [comment]
@banmoy Here:
```
CREATE TABLE IF NOT EXISTS `default_catalog`.`staging`.`accounts_src` (
  `id` BIGINT NOT NULL,
  `email` STRING NULL,
  `phone` STRING NOT NULL,
  `displayname` STRING NOT NULL,
  `password` STRING NULL
  PRIMARY KEY(`id`)
 NOT ENFORCED
) with (
  'server-id' = '5501',
  'connector' = 'mysql-cdc',
  'hostname' = 'xxx',
  'port' = '3306',
  'username' = 'xxx',
  'password' = 'xxx',
  'database-name' = 'staging',
  'table-name' = 'accounts'
);

CREATE TABLE IF NOT EXISTS `default_catalog`.`staging`.`accounts_sink` (
  `id` BIGINT NOT NULL,
  `email` STRING NULL,
  `phone` STRING NOT NULL,
  `displayname` STRING NOT NULL,
  `password` STRING NULL
  PRIMARY KEY(`id`)
 NOT ENFORCED
) with (
  'sink.properties.strip_outer_array' = 'true',
  'password' = '',
  'load-url' = 'starrockscluster-sample2-fe-service.default:8030',
  'username' = 'root',
  'sink.properties.format' = 'json',
  'sink.buffer-flush.interval-ms' = '15000',
  'jdbc-url' = 'jdbc:mysql://starrockscluster-sample2-fe-service.default:9030',
  'connector' = 'starrocks',
  'database-name' = 'staging',
  'table-name' = 'accounts',
  'sink.max-retries' = '10'
);
```

Created using SMT tool with the addition of `server-id` on the cdc connector

**Comment by adhasahar97** [comment]
@banmoy I receive the `DELETE` statement based on the log:
```
-D[6956, 2687, 55837, activity, 128493, 6389c098e1464aebd402fe0b, null, 2022-12-08T09:15:56, 2022-12-08T09:15:56]
2022-12-08 09:27:35,973 INFO  com.starrocks.data.load.stream.StreamTableRegion             [] - Write EOF
2022-12-08 09:27:35,973 INFO  com.starrocks.data.load.stream.StreamTableRegion             [] - uk : staging-contact_mapping, label : ad36faa1-ca44-40ad-a9a8-5b885044d88d commit
2022-12-08 09:27:35,974 INFO  com.starrocks.data.load.stream.DefaultStreamLoader           [] - Stream loading, label : ad36faa1-ca44-40ad-a9a8-5b885044d88d, region : staging-contact_mapping
2022-12-08 09:27:35,975 INFO  com.starrocks.data.load.stream.DefaultStreamLoader           [] - Stream loading, label : ad36faa1-ca44-40ad-a9a8-5b885044d88d, request : PUT http://starrockscluster-sample2-fe-service.default:8030/api/staging/contact_mapping/_stream_load HTTP/1.1
2022-12-08 09:27:35,984 INFO  com.starrocks.data.load.stream.StreamTableRegion             [] - Read EOF
2022-12-08 09:27:35,984 INFO  com.starrocks.data.load.stream.http.StreamLoadEntity         [] - Entity write end, contentLength : -1, total : 197
2022-12-08 09:27:36,016 INFO  com.starrocks.data.load.stream.DefaultStreamLoader           [] - Stream load completed, label : ad36faa1-ca44-40ad-a9a8-5b885044d88d, database : staging, table : contact_mapping, body : {
    "TxnId": 19691,
    "Label": "ad36faa1-ca44-40ad-a9a8-5b885044d88d",
    "Status": "Success",
    "Message": "OK",
    "NumberTotalRows": 1,
    "NumberLoadedRows": 1,
    "NumberFilteredRows": 0,
    "NumberUnselectedRows": 0,
    "LoadBytes": 197,
    "LoadTimeMs": 34,
    "BeginTxnTimeMs": 0,
    "StreamLoadPlanTimeMs": 0,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 14,
    "CommitAndPublishTimeMs": 19
}

```

The first line is the log from the print connector. Followed is the log I received from the starrocks connector

**Comment by banmoy** [comment]
@adhasahar97 This problem will be fixed in #158. Currently you can  use the csv format for workaround, and just remove the `'sink.properties.format' = 'json'`.

**Comment by adhasahar97** [comment]
Hi @banmoy. Using `'sink.properties.format' =' csv'` caused another issue:
```
Caused by: java.lang.RuntimeException: com.starrocks.data.load.stream.exception.StreamLoadFailException: {
    "TxnId": 46420,
    "Label": "ff11b302-f5dd-4886-962b-1832df9ba4ae",
    "Status": "Fail",
    "Message": "Syntax error in line 1:\n...tid,refid,templateid,type,create,edit,view,created_by,...\n                             ^\nEncountered: CREATE\nExpected: CREATE is keyword, maybe `CREATE`\n",
    "NumberTotalRows": 0,
    "NumberLoadedRows": 0,
    "NumberFilteredRows": 0,
    "NumberUnselectedRows": 0,
    "LoadBytes": 0,
    "LoadTimeMs": 0,
    "BeginTxnTimeMs": 0,
    "StreamLoadPlanTimeMs": 1,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 0,
    "CommitAndPublishTimeMs": 0
}
```

**Comment by adhasahar97** [comment]
More log:
```
2022-12-09 02:54:24,863 INFO  org.apache.flink.runtime.executiongraph.ExecutionGraph       [] - Source: table_1_src[97] -> ConstraintEnforcer[98] -> Sink: table_1_sink[98] (1/1) (8a598b4550a1f3c2bc87451351ff8da1) switched from RUNNING to FAILED on basic-session-taskmanager-1-1 @xxx.compute.internal (dataPort=33177).
java.lang.RuntimeException: com.starrocks.data.load.stream.exception.StreamLoadFailException: {
    "TxnId": 46420,
    "Label": "ff11b302-f5dd-4886-962b-1832df9ba4ae",
    "Status": "Fail",
    "Message": "Syntax error in line 1:\n...tid,refid,templateid,type,create,edit,view,created_by,...\n                             ^\nEncountered: CREATE\nExpected: CREATE is keyword, maybe `CREATE`\n",
    "NumberTotalRows": 0,
    "NumberLoadedRows": 0,
    "NumberFilteredRows": 0,
    "NumberUnselectedRows": 0,
    "LoadBytes": 0,
    "LoadTimeMs": 0,
    "BeginTxnTimeMs": 0,
    "StreamLoadPlanTimeMs": 1,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 0,
    "CommitAndPublishTimeMs": 0
}
	at com.starrocks.data.load.stream.DefaultStreamLoadManager.AssertNotException(DefaultStreamLoadManager.java:333) ~[?:?]
	at com.starrocks.data.load.stream.DefaultStreamLoadManager.write(DefaultStreamLoadManager.java:195) ~[?:?]
	at com.starrocks.connector.flink.table.sink.StarRocksDynamicSinkFunctionV2.invoke(StarRocksDynamicSinkFunctionV2.java:152) ~[?:?]
	at org.apache.flink.table.runtime.operators.sink.SinkOperator.processElement(SinkOperator.java:65) ~[flink-table-runtime-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.pushToOperator(CopyingChainingOutput.java:82) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:57) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:29) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:56) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.api.operators.CountingOutput.collect(CountingOutput.java:29) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.table.runtime.util.StreamRecordCollector.collect(StreamRecordCollector.java:44) ~[flink-table-runtime-1.15.3.jar:1.15.3]
	at org.apache.flink.table.runtime.operators.sink.ConstraintEnforcer.processElement(ConstraintEnforcer.java:247) ~[flink-table-runtime-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.pushToOperator(CopyingChainingOutput.java:82) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:57) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.CopyingChainingOutput.collect(CopyingChainingOutput.java:29) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.SourceOperatorStreamTask$AsyncDataOutputToOutput.emitRecord(SourceOperatorStreamTask.java:313) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.api.operators.source.SourceOutputWithWatermarks.collect(SourceOutputWithWatermarks.java:110) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.api.operators.source.SourceOutputWithWatermarks.collect(SourceOutputWithWatermarks.java:101) ~[flink-dist-1.15.3.jar:1.15.3]
	at com.ververica.cdc.connectors.mysql.source.reader.MySqlRecordEmitter$OutputCollector.collect(MySqlRecordEmitter.java:143) ~[?:?]
	at com.ververica.cdc.debezium.table.RowDataDebeziumDeserializeSchema.emit(RowDataDebeziumDeserializeSchema.java:157) ~[?:?]
	at com.ververica.cdc.debezium.table.RowDataDebeziumDeserializeSchema.deserialize(RowDataDebeziumDeserializeSchema.java:124) ~[?:?]
	at com.ververica.cdc.connectors.mysql.source.reader.MySqlRecordEmitter.emitElement(MySqlRecordEmitter.java:118) ~[?:?]
	at com.ververica.cdc.connectors.mysql.source.reader.MySqlRecordEmitter.emitRecord(MySqlRecordEmitter.java:100) ~[?:?]
	at com.ververica.cdc.connectors.mysql.source.reader.MySqlRecordEmitter.emitRecord(MySqlRecordEmitter.java:54) ~[?:?]
	at org.apache.flink.connector.base.source.reader.SourceReaderBase.pollNext(SourceReaderBase.java:143) ~[flink-connector-files-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.api.operators.SourceOperator.emitNext(SourceOperator.java:385) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.io.StreamTaskSourceInput.emitNext(StreamTaskSourceInput.java:68) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.io.StreamOneInputProcessor.processInput(StreamOneInputProcessor.java:65) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.processInput(StreamTask.java:519) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.mailbox.MailboxProcessor.runMailboxLoop(MailboxProcessor.java:203) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.runMailboxLoop(StreamTask.java:807) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.streaming.runtime.tasks.StreamTask.invoke(StreamTask.java:756) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.runtime.taskmanager.Task.runWithSystemExitMonitoring(Task.java:948) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.runtime.taskmanager.Task.restoreAndInvoke(Task.java:927) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.runtime.taskmanager.Task.doRun(Task.java:741) ~[flink-dist-1.15.3.jar:1.15.3]
	at org.apache.flink.runtime.taskmanager.Task.run(Task.java:563) ~[flink-dist-1.15.3.jar:1.15.3]
	at java.lang.Thread.run(Unknown Source) ~[?:?]
Caused by: com.starrocks.data.load.stream.exception.StreamLoadFailException: {
    "TxnId": 46420,
    "Label": "ff11b302-f5dd-4886-962b-1832df9ba4ae",
    "Status": "Fail",
    "Message": "Syntax error in line 1:\n...tid,refid,templateid,type,create,edit,view,created_by,...\n                             ^\nEncountered: CREATE\nExpected: CREATE is keyword, maybe `CREATE`\n",
    "NumberTotalRows": 0,
    "NumberLoadedRows": 0,
    "NumberFilteredRows": 0,
    "NumberUnselectedRows": 0,
    "LoadBytes": 0,
    "LoadTimeMs": 0,
    "BeginTxnTimeMs": 0,
    "StreamLoadPlanTimeMs": 1,
    "ReadDataTimeMs": 0,
    "WriteDataTimeMs": 0,
    "CommitAndPublishTimeMs": 0
}
	at com.starrocks.data.load.stream.DefaultStreamLoader.send(DefaultStreamLoader.java:270) ~[?:?]
	at com.starrocks.data.load.stream.DefaultStreamLoader.lambda$send$2(DefaultStreamLoader.java:113) ~[?:?]
	at java.util.concurrent.FutureTask.run(Unknown Source) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) ~[?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) ~[?:?]
	... 1 more
2022-12-09 02:54:24,883 INFO  org.apache.flink.runtime.source.coordinator.SourceCoordinator [] - Removing registered reader after failure for subtask 0 of source Source: table_1_src[97].
```

SHOW CREATE TABLE in MySQL
```
CREATE TABLE `table_1` (
  `id` int(11) NOT NULL AUTO_INCREMENT,
  `parentid` int(11) DEFAULT NULL COMMENT 'parentid of settings',
  `refid` int(11) DEFAULT NULL COMMENT 'profileid or deptid',
  `templateid` varchar(255) DEFAULT NULL COMMENT 'templateid for settings',
  `type` varchar(255) DEFAULT NULL COMMENT '''user'' or ''department''',
  `create` int(11) DEFAULT '1' COMMENT '0-default,1-allow,2-create',
  `edit` int(11) DEFAULT '1' COMMENT '0-default,1-allow,2-create',
  `view` int(11) DEFAULT '1' COMMENT '0-default,1-allow,2-create',
  `created_by` int(11) DEFAULT NULL COMMENT 'profileid created',
  `modified_by` int(11) DEFAULT NULL COMMENT 'last profileid modified',
  `created_datetime` datetime DEFAULT NULL COMMENT 'settings created date',
  `modified_datetime` datetime DEFAULT NULL COMMENT 'settings modified date'
  PRIMARY KEY (`id`)
) ENGINE=InnoDB AUTO_INCREMENT=11194 DEFAULT CHARSET=latin1
```

SHOW CREATE TABLE in StarRocks:
```
CREATE TABLE `table_1` (
  `id` int(11) NOT NULL COMMENT "",
  `parentid` int(11) NULL COMMENT "parentid of settings",
  `refid` int(11) NULL COMMENT "profileid or deptid",
  `templateid` varchar(65533) NULL COMMENT "templateid for settings",
  `type` varchar(65533) NULL COMMENT "'user' or 'department'",
  `create` int(11) NULL DEFAULT "1" COMMENT "0-default,1-allow,2-create",
  `edit` int(11) NULL DEFAULT "1" COMMENT "0-default,1-allow,2-create",
  `view` int(11) NULL DEFAULT "1" COMMENT "0-default,1-allow,2-create",
  `created_by` int(11) NULL COMMENT "profileid created",
  `modified_by` int(11) NULL COMMENT "last profileid modified",
  `created_datetime` datetime NULL COMMENT "settings created date",
  `modified_datetime` datetime NULL COMMENT "settings modified date"
) ENGINE=OLAP 
PRIMARY KEY(`id`)
COMMENT "OLAP"
DISTRIBUTED BY HASH(`id`) BUCKETS 1 
PROPERTIES (
"replication_num" = "3",
"in_memory" = "false",
"storage_format" = "DEFAULT",
"enable_persistent_index" = "false"
);
```

**Comment by banmoy** [comment]
@adhasahar97 `Syntax error` is fixed in #173. Please use the newest main branch to verify whether your problem is solved. Thanks

**Comment by adhasahar97** [comment]
@banmoy alright thanks!
